### ISSUE TITLE
Looping Named Buckets

### ISSUE THREAD

**Issue by RobinRieger** [elastic/elasticsearch-net#1742]
Is there a way in 2.0 to loop over named buckets (aggregations)?

In 1.7 I was able to do the following 

    foreach (var row in ((FiltersBucket) searchResultForTimePeriod.Aggregations["repeatBuyers"]).Aggregations.Aggregations) {}

The search looked something like this:

```
Filters
    ("repeatBuyers",
        fil => fil.Filters
            (f => f.Name(((int) RepeatBuyersGrouping.One).ToString())
                .Range
                    (ra => ra.OnField(of => of.TotalOrders)
                        .LowerOrEquals(1)),
            f => f.Name(((int) RepeatBuyersGrouping.TwoFive).ToString())
                .Range
                    (ra => ra.OnField(of => of.TotalOrders)
                        .Greater(1)
                        .LowerOrEquals(5)),
...
```

This way I could use the bucket name and the value in the output. I.e. The name of each bucket is a integer and it maps to an enum which I can then use to get the proper 'display' text' for the output. 

With 2.0, I can't seem to loop the 'NamedBucket'. Do I need to call them separately now or was this accidentally dropped?

**Comment by russcam** [comment]
Here's how you would do it in NEST 2.x

```c#
var names = new[] {
	((int)RepeatBuyersGrouping.One).ToString(),
	((int)RepeatBuyersGrouping.TwoFive).ToString()
};

var response = client.Search<Example>(s => s
	.Aggregations(aggs => aggs
		.Filters("repeatBuyers", fil => fil
			.NamedFilters(f => f
				.Filter(names[0], q => q
					.Range(ra => ra
						.Field(of => of.TotalOrders)
						.LessThanOrEquals(1)))
				.Filter(names[1], q => q
					.Range(ra => ra
						.Field(of => of.TotalOrders)
						.GreaterThan(1)
						.LessThanOrEquals(5))
				)
			)
		)
	)
);

var filters = response.Aggs.Filters("repeatBuyers");
foreach (var name in names)
{
	var filter = filters.NamedBucket(name);
	// do something with filter
}
```

The `Aggegations` property on `FiltersBucket` is `private` - @Mpdreamz / @gmarz, is this an intentional change?

**Comment by RobinRieger** [comment]
To clarify, what you are suggesting is to no longer loop the buckets as such but the array I have that I used to make the different buckets. I tried your code and it seems to work (will check more over coming days)

I feel like the `Aggegations`/`Items` being a public property could be useful. I can't see any obvious reason to restrict them..... :confused:... definitely looking forward to hearing more on this.

Thanks!

**Comment by RobinRieger** [comment]
As a quick follow up, I wanted to test making these public, so I compiled the repo locally and changed just the [FiltersBucket](https://github.com/elastic/elasticsearch-net/blob/82c938893b2ff4ddca03a8e977ad14a16da712ba/src/Nest/Aggregations/Bucket/Filters/FiltersBucket.cs) setting the `Aggregations` and `Items` to be public. My code then all compiled again and seems to be working as expected.

**Comment by russcam** [comment]
@RobinRieger cool! Feel free to submit a PR with the change  :+1:

**Comment by gmarz** [comment]
Closed via #1752
